cmake/kokkos_arch.cmake

```diff
@@ -0,0 +1,423 @@
+    LIST(APPEND NEW_LINK_OPTIONS ${FLAGS})
+    GLOBAL_SET(KOKKOS_LINK_OPTIONS ${NEW_LINK_OPTIONS})
+  ENDIF()
+ENDFUNCTION()
+
+# Make sure devices and compiler ID are done
+KOKKOS_CFG_DEPENDS(ARCH COMPILER_ID)
+KOKKOS_CFG_DEPENDS(ARCH DEVICES)
+KOKKOS_CFG_DEPENDS(ARCH OPTIONS)
+
+
+
+
+#-------------------------------------------------------------------------------
+# List of possible host architectures.
+#-------------------------------------------------------------------------------
+SET(KOKKOS_ARCH_LIST)
+LIST(APPEND KOKKOS_ARCH_LIST
+     NONE            # No architecture optimization
+     AMDAVX          # (HOST) AMD chip
+     ARMV80          # (HOST) ARMv8.0 Compatible CPU
+     ARMV81          # (HOST) ARMv8.1 Compatible CPU
+     ARMV8_THUNDERX  # (HOST) ARMv8 Cavium ThunderX CPU
+     ARMV8_TX2       # (HOST) ARMv8 Cavium ThunderX2 CPU
+     WSM             # (HOST) Intel Westmere CPU
+     SNB             # (HOST) Intel Sandy/Ivy Bridge CPUs
+     HSW             # (HOST) Intel Haswell CPUs
+     BDW             # (HOST) Intel Broadwell Xeon E-class CPUs
+     SKX             # (HOST) Intel Sky Lake Xeon E-class HPC CPUs (AVX512)
+     KNC             # (HOST) Intel Knights Corner Xeon Phi
+     KNL             # (HOST) Intel Knights Landing Xeon Phi
+     BGQ             # (HOST) IBM Blue Gene Q
+     POWER7          # (HOST) IBM POWER7 CPUs
+     POWER8          # (HOST) IBM POWER8 CPUs
+     POWER9          # (HOST) IBM POWER9 CPUs
+     KEPLER          # (GPU) NVIDIA Kepler default (generation CC 3.5)
+     KEPLER30        # (GPU) NVIDIA Kepler generation CC 3.0
+     KEPLER32        # (GPU) NVIDIA Kepler generation CC 3.2
+     KEPLER35        # (GPU) NVIDIA Kepler generation CC 3.5
+     KEPLER37        # (GPU) NVIDIA Kepler generation CC 3.7
+     MAXWELL         # (GPU) NVIDIA Maxwell default (generation CC 5.0)
+     MAXWELL50       # (GPU) NVIDIA Maxwell generation CC 5.0
+     MAXWELL52       # (GPU) NVIDIA Maxwell generation CC 5.2
+     MAXWELL53       # (GPU) NVIDIA Maxwell generation CC 5.3
+     PASCAL60        # (GPU) NVIDIA Pascal generation CC 6.0
+     PASCAL61        # (GPU) NVIDIA Pascal generation CC 6.1
+     VOLTA70         # (GPU) NVIDIA Volta generation CC 7.0
+     VOLTA72         # (GPU) NVIDIA Volta generation CC 7.2
+     TURING75         # (GPU) NVIDIA Turing generation CC 7.5
+     RYZEN
+     EPYC
+     KAVERI
+     CARRIZO
+     FIJI
+     VEGA
+     GFX901
+    )
+SET(tmpr "\n       ")
+STRING(REPLACE ";" ${tmpr} KOKKOS_INTERNAL_ARCH_DOCSTR "${KOKKOS_ARCH_LIST}")
+SET(KOKKOS_INTERNAL_ARCH_DOCSTR "${tmpr}${KOKKOS_INTERNAL_ARCH_DOCSTR}")
+
+
+# Setting this variable to a value other than "None" can improve host
+# performance by turning on architecture specific code.
+# NOT SET is used to determine if the option is passed in.  It is reset to
+# default "None" down below.
+KOKKOS_OPTION(ARCH "NONE" STRING "Optimize for specific host architecture. Options are: ${KOKKOS_INTERNAL_ARCH_DOCSTR}")
```

Comment:
It seems that ${KOKKOS_INTERNAL_ARCH_DOCSTR} is multiline string `CMake` does not handle properly. `CMakeCache.txt` shows this option as
```
//Optimize for specific host architecture. Options are:
//\n       NONE
//\n       AMDAVX
//\n       ARMV80
//\n       ARMV81
//\n       ARMV8_THUNDERX
//\n       ARMV8_TX2
//\n       WSM
//\n       SNB
//\n       HSW
//\n       BDW
//\n       SKX
//\n       KNC
//\n       KNL
//\n       BGQ
//\n       POWER7
//\n       POWER8
//\n       POWER9
//\n       KEPLER
//\n       KEPLER30
//\n       KEPLER32
//\n       KEPLER35
//\n       KEPLER37
//\n       MAXWELL
//\n       MAXWELL50
//\n       MAXWELL52
//\n       MAXWELL53
//\n       PASCAL60
//\n       PASCAL61
//\n       VOLTA70
//\n       VOLTA72
//\n       TURING75
//\n       RYZEN
//\n       EPYC
//\n       KAVERI
//\n       CARRIZO
//\n       FIJI
//\n       VEGA
//\n       GFX901
```
but `ccmake` only shows the first line: "Optimize for specific host architecture. Options are:"
Of course, you see all the options if switch into `help` mode. It would be helpful to explicitly tell the user to switch into that mode here or have all the options in one line.

Reply:
I think we should switch to a single line. When CMake prints long strings, it already does a "pretty print" and puts in line breaks